src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1475,3 +1561,345 @@ function writeManifestFile(
+        return { status: 'abort' }
+      }
+
+      const match = output.match(
+        /Blocks: (\d+)\/(\d+), Data Downloaded: ([\d.]+)/
+      )
+      if (match) {
+        downloadedBlocks = parseInt(match[1], 10)
+        totalBlocks = parseInt(match[2], 10)
+        downloadedData = parseInt(match[3])
+
+        const percent = (downloadedBlocks / totalBlocks) * 100
+        const currentTime = Date.now()
+        const elapsedTime = (currentTime - startTime) / 1000 // in seconds
+        const downloadedDataInMiB = downloadedData / 1024 / 1024
+        const downloadSpeed = downloadedData / elapsedTime
+        const totalSize = blockSize * totalBlocks
+        const eta = calculateEta(downloadedData, downloadSpeed, totalSize) ?? 0
+
+        sendFrontendMessage('gameStatusUpdate', {
+          appName,
+          status: 'patching',
+          runner: 'hyperplay',
+          folder: gameInfo.install.install_path
+        })
+
+        window?.webContents.send(`progressUpdate-${appName}`, {
+          appName,
+          status: 'patching',
+          runner: 'hyperplay',
+          folder: gameInfo.install.install_path,
+          progress: {
+            folder: gameInfo.install.install_path,
+            totalSize,
+            percent,
+            diskSpeed: downloadSpeed / 1024 / 1024,
+            downSpeed: downloadSpeed / 1024 / 1024,
+            bytes: downloadedDataInMiB,
+            eta
+          }
+        })
+
+        logInfo(
+          `Progress: ${percent.toFixed(2)}%, Downloaded: ${getFileSize(
+            downloadedData
+          )}, Speed: ${getFileSize(
+            downloadSpeed
+          )}/s, ETA: ${eta} totalSize: ${totalSize} = ${getFileSize(
+            totalSize
+          )}`,
+          LogPrefix.HyperPlay
+        )
+      }
+    }
+  } catch (error) {
+    logError(`Error while patching ${error}`, LogPrefix.HyperPlay)
+
+    trackEvent({
+      event: 'Patching Failed',
+      properties: {
+        error: `${error}`,
+        game_name: gameInfo.app_name,
+        game_title: gameInfo.title,
+        platform: getPlatformName(platform),
+        platform_arch: platform
+      }
+    })
+
+    captureException(new Error(`Error while patching ${error}`), {
+      extra: {
+        method: 'applyPatching',
+        appName,
+        title: gameInfo.title,
+        install_path,
+        platform,
+        version,
+        newVersion
+      }
+    })
+
+    return { status: 'error', error: `Error while patching ${error}` }
+  }
+  // need this to cover 100% of abort cases
+  if (aborted) {
+    return { status: 'abort' }
+  }
+
+  trackEvent({
+    event: 'Patching Success',
+    properties: {
+      game_name: gameInfo.app_name,
+      game_title: gameInfo.title,
+      platform: getPlatformName(platform),
+      platform_arch: platform
+    }
+  })
+
+  logInfo(`Patching ${appName} completed`, LogPrefix.HyperPlay)
+  return { status: 'done' }
```

Comment:
couldn't this be moved into the try block above? It would be closer to the code's context there

Reply:
It cannot go above the `if (aborted)`. I needed this check here to cover some edge cases on Windows.
Linux and macOS deal fine with the AbortController but Windows needs more checks otherwise it will keep going to the `done` status and the game will have the info updated without being updated.

Reply:
yeah i was thinking the `if (aborted)` would go in the try block too

Reply:
Ah I see now. Will do